pywinauto/controls/common_controls.py

```diff
@@ -3388,6 +3388,102 @@ def set_view(self, viewType):
         if res == 0:
             raise RuntimeError('Failed to set view in Calendar')
 
+    # ----------------------------------------------------------------
+    def set_id(self, ID):
+        """
+        Set the calendar type.
+
+        Receive only one parameter, which takes variants below:
+        'gregorian', 'gregorian_us', 'japan', 'taiwan', 'korea',
+        'hijri', 'thai', 'hebrew', 'gregorian_me_french',
+        'gregorian_arabic', 'gregorian_english_xlit',
+        'gregorian_french_xlit', 'umalqura'
+        """
+
+        dict_types = {
+            'gregorian': win32defines.CAL_GREGORIAN,
+            'gregorian_us': win32defines.CAL_GREGORIAN_US,
+            'japan': win32defines.CAL_JAPAN,
+            'taiwan': win32defines.CAL_TAIWAN,
+            'korea': win32defines.CAL_KOREA,
+            'hijri': win32defines.CAL_HIJRI,
+            'thai': win32defines.CAL_THAI,
+            'hebrew': win32defines.CAL_HEBREW,
+            'gregorian_me_french': win32defines.CAL_GREGORIAN_ME_FRENCH,
+            'gregorian_arabic': win32defines.CAL_GREGORIAN_ARABIC,
+            'gregorian_english_xlit': win32defines.CAL_GREGORIAN_XLIT_ENGLISH,
+            'gregorian_french_xlit': win32defines.CAL_GREGORIAN_XLIT_FRENCH,
+            'umalqura': win32defines.CAL_UMALQURA
+        }
+        if ID in dict_types:
+            self.send_message(win32defines.MCM_SETCALID, dict_types[ID], 0)
+        else:
+            raise RuntimeError('You typed incorrect type of Calendar')

```

Comment:
Should be `ValueError('Incorrect calendar ID (use one of {0})'.format(dict_types.keys()))`